File: Shanjay K/Lab 2/javascript/javascript_playground/kafka/topic.js
//const Kafka = require("kafkajs").Kafka
const {Kafka} = require("kafkajs")

run();
async function run(){
    try{
        const kafka = new Kafka({
            "cliendId": "myapp",
            "brokers": ["localhost:9093"]
        })

        const admin = kafka.admin();
        console.log("Connecting...")
        await admin.connect()
        console.log("Connected!")
        await admin.createTopics({
            "topics": [{
                "topic" : "Users",
                "numPartitions": 2
            }]
        })
        console.log("Created Successfully!")
        await admin.disconnect();

    }
    catch(ex)
    {
        console.error(`Error: ${ex}`)
    }
    finally {
        process.exit(0);
    }
}

File: Yusuf Shaik/Kafka Python/topic.py
from kafka import KafkaAdminClient
from kafka.admin import NewTopic

try:
    admin = KafkaAdminClient(bootstrap_servers=['localhost:9093', 'localhost:9094', 'localhost:9095'])

    topic = NewTopic(name='Users', num_partitions=2, replication_factor=2)
    admin.create_topics([topic])
    print("Created topic")
except Exception:
    print("Failed to create topic")


